### PR TITLE
[Miyo-only search] - Step 6: Stop mirroring QA exclusions into Miyo

### PR DESCRIPTION
**Pull request by zeroliu**
Relates to Brevilabs/obsidian-copilot-private#204

## Why

Copilot tries to keep Miyo's folder exclusions matching its own, and the repair path is destructive. When the two drift, the Miyo settings tab shows a warning: "Miyo's excluded folders don't match your Copilot folder — chats could be indexed and exposed. Resync to update them." Pressing **Resync Miyo** deletes the folder registration and re-creates it, which discards the fine-grained write filters and grants the user set in the Miyo app. Changing the Copilot folder also raises a startup notice, "Miyo search needs a resync — open the Miyo settings tab." Copilot decides all of this from `miyoSyncedExclusions`, a device-local receipt of what it last sent, which a multi-device sync or a Reset Settings can leave wrong in either direction.

Part of what Copilot mirrors is `qaInclusions` / `qaExclusions`. Freezing a snapshot of those rules into Miyo at every change is what makes drift detection necessary in the first place: Miyo cannot edit what Copilot keeps overwriting, and Copilot cannot tell whether the user changed the scope in the Miyo app on purpose. Narrowing what Miyo indexes belongs to Miyo, where it can be changed after registration.

## What

| Behavior | Before | After |
| --- | --- | --- |
| Miyo folder rules after registration | Resync deletes and re-creates the registration, dropping the write filters and grants set in the Miyo app | Copilot never mutates an existing registration; the banner, the **Resync Miyo** button, and the startup notice are gone, and Miyo owns its scope |
| Copilot scope reaching Miyo | Re-sent on every root or rule change, tracked by a device-local `miyoSyncedExclusions` receipt | Sent once at initial registration; the receipt has no reader and is stripped from `data.json` on every load and save |
| Initial registration payload | Copilot's working roots, Obsidian's ignored paths, and the user's `qaInclusions` / `qaExclusions` | Copilot's working roots and Obsidian's ignored paths only, normalized so empty entries and `.` / `..`, which Miyo can read as the vault root or its parent, are dropped |
| `qaInclusions` / `qaExclusions` locally | Scope vault search, grep, the agent's file-tree listing, the Miyo retriever, and Relevant Notes | Unchanged |
| Vault search candidate window | Miyo's 1,000 maximum when local rules or a stale scope could drop candidates, otherwise a 2x margin | Always the 1,000 maximum, because Copilot no longer defines the server's scope and cannot tell whether a ranked prefix is content its local filter is about to drop |
| Relevant Notes after a Copilot root change | Keeps showing notes under the newly protected root until an unrelated refresh | Refreshes immediately |
| Registering while the plugin unloads | A settings tree outliving its plugin could register the old vault and enable Miyo in the next lifecycle | Registration rights are revoked at unload and the request reads live settings at its boundary |

> **Before:** a user tunes their vault's write filters in the Miyo app, then changes Copilot's folder in Obsidian. They get a banner saying Miyo's exclusions no longer match, and its **Resync Miyo** button deletes the registration and takes those filters with it.
>
> **After:** no banner appears, Copilot leaves the registration alone, and the Miyo app stays the only place that folder's scope is set.

## Non goal

- Changing what `qaInclusions` / `qaExclusions` scope inside Copilot. They keep filtering vault search, grep, the agent file-tree listing, the Miyo retriever, and Relevant Notes exactly as they do today, so a vault carrying legacy rules sees no difference. The editable access scope arrives with Brevilabs/obsidian-copilot-private#358.
- Retroactively removing exclusions an earlier build already pushed into Miyo. A vault registered before this keeps whatever Miyo holds; a vault registered after it sets its server-side scope in the Miyo app.
- Sending any Copilot scope to Miyo after initial registration, or changing Miyo's own folder rules and Relay access.
- Gating Relevant Notes on the future access scope; it shows what Miyo indexed, filtered by today's local rules.
- Grouping Miyo's related-search results by note before its result limit. That belongs in the Miyo backend, not in a client-side over-fetch.

## Screenshot

No image. The removed surfaces render only when Copilot's receipt disagrees with a live registration, a state that needs a real Miyo instance and a drifted record to stage, so a before/after pair cannot be captured deterministically. The exact removed copy is the evidence:

- Miyo settings banner: "Miyo's excluded folders don't match your Copilot folder — chats could be indexed and exposed. Resync to update them.", with a **Resync Miyo** button, or "Remove and re-add this folder in the Miyo app." on a remote or mobile connection.
- Startup notice and migration-summary row: "Miyo search needs a resync — open the Miyo settings tab."

Nothing else in the Miyo tab moves; the remaining rows keep their order and copy.

## Risk

**High**

| Criterion | Status | Reason |
| --- | :---: | --- |
| No behavior change, or a cosmetic/copy/docs/config change visible where it renders, or deterministic tests cover the changed behavior | ✅ | Tests cover the registration payload and its normalization, lifecycle expiry at the request boundary, the receipt's removal at load and save, the Miyo retriever's candidate window and scope filter, and root-change invalidation of Relevant Notes |
| A defect would fail CI or be obvious on first use | ❌ | Every test mocks Miyo, so the live registration payload and the 1,000-result request are only exercised against a real instance |
| A revert fully restores prior state, including persisted data | ❌ | `miyoSyncedExclusions` is stripped from `data.json` on every load and save; a revert cannot restore the receipt, and the first run after it would offer a resync again |
| No auth, permissions, secrets, or input-handling surface changes | ❌ | What a newly registered vault exposes to Miyo widens: folders named by `qaExclusions` are no longer excluded at registration, so Miyo indexes them and any enabled Relay client can read them |
| No public API, plugin API, message, or on-disk contract changes | ❌ | A persisted settings field is removed; the `index-vault-to-copilot-index` command id is deliberately preserved |
| No core-path concurrency, async-lifecycle, or state-machine changes | ❌ | Vault registration is now guarded by plugin lifecycle at the request boundary and before every settings write |
| No hot-path behavior lacks deterministic coverage | ✅ | Retriever and Relevant Notes tests cover the scope filter, search-all ownership, the candidate window, and the excluded-root cases |
| No new dependency | ✅ | Dependency manifests are unchanged |
| Human-only behavior stays in one feature area and surfaces quickly | ✅ | The change is confined to the Miyo integration; a wrong registration payload is visible in the Miyo app on the next registration, and a wrong candidate window in the next vault search |

Review: inspect `registerVault` in `src/settings/v2/components/MiyoSettings.tsx`, `searchMiyo` and `filterByCopilotPatterns` in `src/search/miyo/MiyoSemanticRetriever.ts`, `cleanupLegacyFields` in `src/services/settingsSecretTransforms.ts`, and `onunload` plus `isPluginLifecycleActive` in `src/main.ts` line by line. Then run Verification steps 2-6, including the reload-mid-registration variant in step 6.

## Verification

1. Load this branch in a vault that is already registered with a local Miyo, and in the Miyo app give that folder a distinctive extra rule, such as a restricted write filter.
2. In **Settings → Copilot → Basic**, change the Copilot folder. Expect no "Miyo search needs a resync" notice, no warning banner or Resync button in the Miyo tab, and, in the Miyo app, the folder's exclusions and write filter exactly as you left them.
3. With a note open, watch the Relevant Notes pane across that root change: rows under the new root disappear without reopening the note.
4. In a second, unregistered vault holding a note in `Private/`, add at least one Obsidian **Excluded files** entry, then quit Obsidian, put `"qaExclusions": "Private"` in that vault's `data.json`, and reopen. Press **Connect** in the Miyo tab and confirm the registration. In the Miyo app, the new folder's exclusions list the Copilot roots and the Obsidian ignored path, with no `Private` entry and no `.` or `..` entry.
5. In that same vault, ask Quick Chat something only the `Private/` note answers. Expect it not to be found, and expect `qaExclusions` to still hold `Private` in `data.json` after the session.
6. Open the Miyo tab in a third vault that is not registered, press **Connect** to reach the add-vault confirmation, then disable and re-enable the plugin from Obsidian's community-plugins pane before confirming. Expect no folder to appear in Miyo and Miyo search to stay off.

## Note

Part of the Brevilabs/obsidian-copilot-private#284 epic, which deliberately does not resurrect the old pattern editors or keep sending Copilot's scope to Miyo.

### CODE REVIEW

**Comment by zeroliu** [comment]
Addressed the connector findings in `03268921`:

- preserved the legacy `index-vault-to-copilot-index` command id;
- confirmed the target branch already removes retired embedding secrets and runs legacy-index cleanup per device;
- applied path-resolvable QA rules to unresolved Miyo results and fail closed for metadata-dependent rules;
- kept Copilot roots out of Miyo at registration, additively reconciles current/historical roots without deleting the folder or replacing Miyo-owned rules, and uses Miyo's full 1,000-result window when user QA rules require local filtering;
- retained Obsidian ignored paths in initial registration.

The full test suite, format, lint, build, and Obsidian review pass on the pushed head.

@codex review

File: src/miyo/miyoSystemExclusions.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  );
+  if (missing.length === 0) return false;
+
+  // Copilot roots only accumulate in copilotRootHistory. Preserve every rule
+  // the user configured in Miyo and append only missing system roots, so a
+  // stale device or overlapping root change cannot erase a narrower scope.
+  // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
+  await client.updateFolder(baseUrl, folderName, {
+    exclude_folders: [...current, ...missing],
+  });
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/miyo/miyoSystemExclusions.ts]: Fixed in c8d8cf04 by removing the GET/PATCH reconciliation path entirely. Mutable Copilot QA scope stays local, so Copilot cannot overwrite concurrent or UI-authored Miyo folder settings.

File: src/settings/v2/components/MiyoSettings.tsx [hunk 2]
```diff
@@ -435,92 +316,33 @@ export const MiyoSettings: React.FC = () => {
...
+          // Remote read (Relay) is enabled by default so a user who turns on
+          // Miyo's Relay connector — itself an explicit, paid, signed-in opt-in
+          // (ChatGPT/Claude linked once) — gets cloud access to this vault without
+          // re-configuring it per folder. It's inert until Relay is actually on, so
+          // it doesn't expose anything on its own; the register modal states this
+          // plainly rather than promising absolute privacy. A user who wants this
+          // vault kept out of Relay can flip allow_remote_read off per folder in Miyo.
+          allow_remote_read: true,
+        },
+        customUrl || undefined
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/settings/v2/components/MiyoSettings.tsx]: Fixed in c8d8cf04. Registration checks plugin lifecycle at the request boundary and again before settings writes or notifications. The regression test expires the lifecycle in beforeRequest and proves no request body, enable setting, or notification is produced.

File: src/miyo/miyoResync.ts [hunk 3]
```diff
@@ -1,774 +0,0 @@
...
-    // opted out in Miyo — the exact privacy hole this resync exists to close.
-    desired.allow_remote_read = record.allow_remote_read === true;
-    desired.allow_writes = record.allow_writes === true;
-    // Same reasoning one level up: the record's own filters are part of the
-    // boundary it enforces, and a rebuild must not come back broader.
-    preserveRecordScope(desired, record);
-
-    // Last point at which abandoning is still safe: once the DELETE is out, the
-    // POST must follow or the vault is left unregistered.
-    await client.deleteFolder(folderName, customUrl, () => assertCurrentLifecycle(lifecycle));
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/miyo/miyoResync.ts:745]: Fixed in c8d8cf04. Removed the unused getFolder, updateFolder, and deleteFolder client surface and their tests. addFolder remains because registration uses its beforeRequest lifecycle guard required by the companion unload finding.

File: src/settings/copilotRootChange.ts [hunk 4]
```diff
@@ -224,4 +225,15 @@ export async function applyCopilotRootChange(app: App, newRoot: string): Promise
...
     setSettings((current) => buildRootPatch(current, folder));
   });
+
+  // Local filtering protects the data immediately. Reconcile Miyo in the
+  // background so the newly protected root also stops consuming its bounded
+  // ranked candidate pool; startup retries this if Miyo is currently offline.
+  // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
+  const updatedSettings = getSettings();
+  if (updatedSettings.enableMiyo) {
+    void syncMiyoSystemExclusions(app, updatedSettings).catch((error) => {
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/settings/copilotRootChange.ts]: Fixed in c8d8cf04. A persisted Copilot root change now emits notifyMiyoIndexChanged even when Miyo is disabled, so Related Notes refreshes under the new root. Both enabled and disabled cases are covered.

File: src/settings/v2/components/MiyoSettings.tsx [hunk 5]
```diff
@@ -435,92 +316,33 @@ export const MiyoSettings: React.FC = () => {
...
-      // resync prompt over a stale scope. Written before the enable step and
-      // regardless of `superseded()`: the server-side registration DID happen
-      // with this exact body, whatever the UI does afterwards.
-      if (submission.created !== null) {
-        updateSetting("miyoSyncedExclusions", submission.receipt);
+      // Keep Copilot's own roots and Obsidian-ignored paths out of Miyo's ranked
+      // candidate pool. They can contain enough chunks to exhaust the server's
+      // bounded result window before Copilot applies its local QA filter.
+      // User-authored QA rules remain local; Miyo owns those folder rules
+      // instead of receiving a snapshot Copilot cannot keep current.
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/settings/v2/components/MiyoSettings.tsx:338]: Fixed in c8d8cf04. Related Notes now applies the existing createCopilotPatternFilter boundary before accepting Miyo results, with a regression test proving QA-excluded notes are omitted.

File: src/miyo/miyoSystemExclusions.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ * array. Each queued call re-reads the folder before merging, so concurrent
+ * root changes cannot overwrite one another or the user's Miyo-owned rules.
+ *
+ * @param app - Active Obsidian app whose vault registration is reconciled.
+ * @param settings - Settings snapshot that triggered the reconciliation.
+ * @returns Whether Miyo's folder record needed an update.
+ */
+export function syncMiyoSystemExclusions(app: App, settings: CopilotSettings): Promise<boolean> {
+  const previous = syncQueues.get(app) ?? Promise.resolve();
+  const next = previous.then(() => reconcileSystemExclusions(app, settings));
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/miyo/miyoSystemExclusions.ts]: Fixed in c8d8cf04 by deleting the module-scoped reconciliation queue along with the reconciler. No queued promise can survive plugin reload and block the new instance.

File: docs/settings.md [hunk 7]
```diff
@@ -183,13 +183,14 @@ Miyo supplies Copilot's current semantic search and selected chat-history search
...
+| **Connector · Relay → Set up in Miyo** | Status detected from Miyo | Opens Miyo's Connector setup. Connector can let cloud ChatGPT or Claude clients read and write registered local files. This is a separate, explicit Miyo Relay setup and privacy boundary.                                                                                              |
+| **Remote Miyo server (advanced)**      | Blank                     | Expand the disclosure and enter a server URL. Blank uses local discovery. The value saves on blur and is used by both Copilot's Miyo connection and the Connector row.                                                                                                                  |
+
+Copilot keeps its current and historical working folders and Obsidian's ignored paths out of Miyo's index, then applies user-authored inclusion and exclusion rules to its own Miyo results at query time. Other per-folder include and exclude rules belong in the Miyo app and also determine what any enabled Relay client can read.
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply docs/settings.md]: Fixed in c8d8cf04. The docs now state that Copilot roots and Obsidian ignored paths are sent only during initial registration, and that Copilot does not overwrite existing or later Miyo-managed exclusions.

File: src/settings/v2/components/MiyoSettings.tsx [hunk 8]
```diff
@@ -435,92 +316,33 @@ export const MiyoSettings: React.FC = () => {
...
+      // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
+      const created = await new MiyoClient({ plusLicenseKey: settings.plusLicenseKey }).addFolder(
+        {
+          path: vaultBase,
+          exclude_folders: [
+            ...new Set([...getSystemExcludedFolders(settings), ...extractAppIgnoreSettings(app)]),
+          ],
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/settings/v2/components/MiyoSettings.tsx]: Fixed in c8d8cf04. Initial-registration normalization drops empty paths plus root and parent pointers, trims only trailing separators, and preserves otherwise inert patterns such as ./notes. Tests cover dot, parent, leading slash, and duplicate-separator inputs.

File: src/search/miyo/MiyoSemanticRetriever.ts [hunk 9]
```diff
@@ -138,23 +149,17 @@ export class MiyoSemanticRetriever extends BaseRetriever {
...
+      // expose pagination and clamps this endpoint at 1,000.
+      // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
       const settings = getSettings();
       const limit =
-        this.returnAll || hasUserQaPatterns(settings) || isMiyoScopeMismatch(this.app, settings)
-          ? RETURN_ALL_LIMIT
-          : Math.min(this.finalK * 2, RETURN_ALL_LIMIT);
+        this.returnAll || hasUserQaPatterns(settings)
+          ? MIYO_SEARCH_CANDIDATE_LIMIT
+          : Math.min(this.finalK * 2, MIYO_SEARCH_CANDIDATE_LIMIT);
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/search/miyo/MiyoSemanticRetriever.ts]: Fixed in c8d8cf04. returnAll alone now requests only the bounded two-times-finalK candidate margin. The 1,000-result window is reserved for local QA filtering or historical custom Copilot roots that can remove ranked candidates.

**Comment by zeroliu** [comment]
Addressed all inline findings in `c8d8cf04`: local QA now covers Related Notes, root changes invalidate it, registration is lifecycle-guarded and normalizes only initial exclusions, `returnAll` keeps a bounded candidate request, and the unsafe Miyo GET/PATCH queue plus dead client methods are removed. All nine threads have replies and are resolved. Format, lint, build, 6,390 tests, and the Obsidian review gate pass on this head.

@codex review

File: src/search/findRelevantNotes.ts [hunk 10]
```diff
@@ -91,7 +93,11 @@ async function searchRelatedNotesWithMiyo(
         continue;
       }
       const relativePath = getVaultRelativeMiyoPath(app, result.path);
-      if (relativePath !== filePath && !scoreByPath.has(relativePath)) {
+      // Related Notes is another Miyo retrieval surface, so it must enforce the
+      // same local QA boundary as chat retrieval. Otherwise a note excluded in
+      // Copilot can still appear in the pane when Miyo returns it.
+      // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
+      if (relativePath !== filePath && !scoreByPath.has(relativePath) && isAllowed(relativePath)) {
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/search/findRelevantNotes.ts]: Fixed in eb636680. Related search now requests a wider ranked window (5x the pane's maximum) and Copilot caps the pane itself after the QA filter, so an eligible note ranked below a run of excluded results still appears. Regression test covers 20 excluded results followed by an allowed one.

File: src/search/miyo/MiyoSemanticRetriever.ts [hunk 11]
```diff
@@ -138,23 +149,21 @@ export class MiyoSemanticRetriever extends BaseRetriever {
...
+      // root. Because Miyo receives roots only during initial registration,
+      // local filtering may need to remove candidates from a root chosen later.
+      // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
       const limit =
-        this.returnAll || hasUserQaPatterns(settings) || isMiyoScopeMismatch(this.app, settings)
-          ? RETURN_ALL_LIMIT
-          : Math.min(this.finalK * 2, RETURN_ALL_LIMIT);
+        hasUserQaPatterns(settings) || getSystemExcludedFolders(settings).length > 1
+          ? MIYO_SEARCH_CANDIDATE_LIMIT
+          : Math.min(this.finalK * 2, MIYO_SEARCH_CANDIDATE_LIMIT);
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/search/miyo/MiyoSemanticRetriever.ts]: Fixed in eb636680. The heuristic is gone: since Copilot no longer defines Miyo's exclusion scope past initial registration, the server scope is never knowable, so vault search always requests Miyo's full exposed candidate pool. That deletes hasUserQaPatterns and the root-count guess rather than replacing one unprovable signal with another.

File: src/miyo/MiyoClient.ts [hunk 12]
```diff
@@ -720,7 +650,7 @@ export class MiyoClient {
     baseUrl: string,
     path: string,
     options: {
-      method: "GET" | "POST" | "DELETE";
+      method: "GET" | "POST" | "PATCH" | "DELETE";
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/miyo/MiyoClient.ts]: Fixed in eb636680. The request helper's method union is back to GET | POST, the only methods any remaining Miyo call uses.

File: src/settings/v2/components/MiyoSettings.tsx [hunk 13]
```diff
@@ -433,101 +319,60 @@ export const MiyoSettings: React.FC = () => {
...
+      // Keep Copilot's own roots and Obsidian-ignored paths out of Miyo's ranked
+      // candidate pool. They can contain enough chunks to exhaust the server's
+      // bounded result window before Copilot applies its local QA filter.
+      // User-authored QA rules remain local; Miyo owns those folder rules
+      // instead of receiving a snapshot Copilot cannot keep current.
+      // https://github.com/Brevilabs/obsidian-copilot-private/issues/284
+      const initialExclusions = [
+        ...new Set(
+          [...getSystemExcludedFolders(settings), ...extractAppIgnoreSettings(app)]
+            .map((folder) => folder.replace(/\\/g, "/").replace(/\/+$/, ""))
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/settings/v2/components/MiyoSettings.tsx:343]: Fixed in eb636680. registerVault reads live settings when registration begins, so the endpoint, credential, and Copilot roots all come from the current configuration rather than the modal's creation-time closure. Regression test changes the Copilot root while the modal is open and proves the new root reaches exclude_folders.

File: src/search/findRelevantNotes.ts [hunk 14]
```diff
@@ -78,6 +79,7 @@ async function searchRelatedNotesWithMiyo(
     }
     const scoreByPath = new Map<string, number>();
     const results = response.results;
+    const isAllowed = createCopilotPatternFilter(app);
```
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/search/findRelevantNotes.ts]: Fixed in eb636680. The Relevant Notes request key now includes the local QA scope (qaInclusions, qaExclusions, the active Copilot root, and its history), so a direct settings replacement retires a settled result without needing an index signal. Regression test replaces qaExclusions and asserts the excluded row disappears.

File: src/search/findRelevantNotes.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (zeroliu) [reply src/search/findRelevantNotes.ts]: Correction to the previous reply: reverted in 7a422b50. Related search keeps requesting the pane's result count. The real cause is that Miyo's related endpoint applies its limit to chunks, so the fix belongs there: Miyo will group related results by note before applying the limit. A client-side over-fetch plus cap would only mask that, and it would multiply the payload of a request that runs on every note switch.

File: src/settings/v2/components/BasicSettings.tsx [hunk 15]
```diff
@@ -160,31 +148,6 @@ export const BasicSettings: React.FC = () => {
       app,
       () => {
         void applyCopilotRootChange(app, folder)
```
Comment (logancyang) [inline src/settings/v2/components/BasicSettings.tsx]: [P1] Keep Miyo scope drift visible after changing the root

Registration sends the current and historical Copilot roots as one-time Miyo exclusions, while this path now changes the root and reports success without reconciling or warning. Miyo therefore keeps the previous server scope, so the new chat folder can be indexed and become Relay-readable without any on-screen notice. Keep at least read-only drift detection with an actionable warning, or update only Copilot-owned exclusions through a concurrency-safe API that preserves Miyo-owned rules.

File: src/search/findRelevantNotes.test.ts [hunk 16]
```diff
@@ -178,6 +182,26 @@ describe("findRelevantNotes", () => {
       expect(result.notes[1].metadata.hasBacklinks).toBe(true);
     });
 
+    it("filters Miyo results through Copilot QA rules (https://github.com/Brevilabs/obsidian-copilot-private/issues/284)", async () => {
```
Comment (logancyang) [inline src/search/findRelevantNotes.test.ts]: [P2] Make this exclusion test exercise an existing file

private/secret.md is never added to filesByPath in the shared fixture, so result resolution drops it even when createCopilotPatternFilter ignores qaExclusions. This test therefore passes while the protection it names is absent. Add the private file to the mocked vault and assert the intended upgrade contract.

File: src/search/searchUtils.ts [hunk 17]
```diff
@@ -59,55 +57,22 @@ export function getDecodedPatterns(value: string): string[] {
...
+ *
+ * @param options - The caller's raw inclusion and exclusion strings.
+ * @returns The categorized rules, each null when the caller has none.
  */
-export function getMatchingPatterns(options?: {
-  inclusions?: string;
-  exclusions?: string;
-  isProject?: boolean;
-}): {
+export function getMatchingPatterns(options?: { inclusions?: string; exclusions?: string }): {
```
Comment (logancyang) [inline src/search/searchUtils.ts]: [P1] Preserve legacy exclusions until the access-scope migration

Vaults configured before the editor was removed still carry intentional qaInclusions and qaExclusions. Returning no vault-wide rules here makes private or journal folders eligible for Quick Chat retrieval, grep, file-tree output, tag/time search, and Relevant Notes, potentially sending their content to a model without user action. Persisting the fields is not enough. Continue applying them until they migrate into the visible access scope, or require explicit consent before widening access.

Comment (zeroliu) [reply src/search/searchUtils.ts]: Fixed in `ab6509e4`, which reverts "Stop reading qaInclusions and qaExclusions" in full.

Both fields scope retrieval again everywhere they did before: `getAllQAMarkdownContent`, `GrepScanner`, `FilterRetriever` (tag/time search), `FileTreeTools`, `MiyoSemanticRetriever.filterByCopilotPatterns`, and `findRelevantNotes`. `getMatchingPatterns()` falls back to the vault-wide rules again and keeps its `isProject` escape hatch, and `createCopilotPatternFilter(app)` takes `app` back along with the path-only/metadata fail-closed handling for paths this vault cannot resolve. A vault carrying legacy rules sees no behavior change from this PR.

What the PR still does is the Miyo half: Copilot no longer mirrors those rules into Miyo's folder registration and no longer keeps a receipt to detect that the server-side scope drifted. `registerVault` sends the vault root, Copilot's own working-folder history, Obsidian's ignored paths, and the Relay grant. User-authored QA rules stay local. A vault already registered with synced exclusions keeps them; a vault registered after this lands scopes its server-side indexing in the Miyo app, where it can be changed after registration instead of being frozen at the first POST.

File: src/search/findRelevantNotes.test.ts (hunk 16 above)
Comment (zeroliu) [reply src/search/findRelevantNotes.test.ts]: Fixed in `0ad19ebf`. You were right that the test was vacuous: `private/secret.md` was never in the fixture's `filesByPath`, so `getAbstractFileByPath` returned null and result resolution dropped the row on its own, independent of the filter.

Registering the path in the shared vault fixture makes the assertion depend on `createCopilotPatternFilter`. Mutation-checked it: stubbing the filter's `inclusions`/`exclusions` to null now fails the test at the `toEqual(["alpha.md"])` assertion, where it passed before.
